### PR TITLE
cleanup: remove seat management

### PR DESCRIPTION
**Pull request by pb82**
Jira Issue: https://issues.redhat.com/browse/AAP-25245

## Description

Removes the seat check logic and relevant unit tests.

## Testing

1. Run the wisdom service locally and try to log in with an AAP licensed user
2. It should succeed

### Scenarios tested

as described above

## Production deployment

- [x] This code change is ready for production on its own
- [ ] This code change requires the following considerations before going to production:

### CODE REVIEW

File: ansible_ai_connect/users/authz_checker.py [hunk 1]
```diff
@@ -1,459 +0,0 @@
...
-        try:
-            for item in result["items"]:
-                if item["role"]["id"] == "OrganizationAdmin":
-                    return True
-        except (KeyError, ValueError):
-            return False
-
-        return False
-
-    def rh_org_has_subscription(self, organization_id: int) -> bool:
```
Comment (goneri) [inline ansible_ai_connect/users/authz_checker.py:366]: We still need this method to know if the Org has an AAP subscription.

**Review by goneri — CHANGES_REQUESTED** [review]

**Comment by pb82** [comment]
@goneri thanks for the review. In that case, I guess we should remove the `check` method of the AMSCheck and CIAMCheck since they are no longer used?

**Comment by goneri** [comment]
That sounds reasonable to me, even if we still use a tiny part the module.

**Comment by pb82** [comment]
Closing this to split it up into multiple PRs

**Pull request closed without merging**